### PR TITLE
Create my_articles feature

### LINKED ISSUE #77: Create a my-articles feature
- [x] Create a new folder under \Smarticle\projboard\templates, there will be located the html parts of this feature
 - [x]  Create the html file and css
 - [x] Add to the \Smarticle\projboard\views.py file a function to render this html part
 - [x] Add to the Smarticle\finalProject_g4\urls.py file a new path to create a new route
 - [x] Add a test file that test the changes you add

### LINKED ISSUE #95: Test returned values in my_articles
After authentication is added, fix the missing test of returned values in `test_client.py`, `my_articles`.

### PR DESCRIPTION
**Pull request by finziamit**
Resolve #77 
Resolve #95

Adding my_articles feature

### CODE REVIEW

File: projboard/tests/test_client.py [hunk 1]
```diff
@@ -94,6 +94,12 @@ def test_fill_article_delete(client):
     assert '404.html' in template_names
 
 
+@pytest.mark.django_db
+def test_get_my_articles(client):
+    response = client.get("/my_articles/")
```
Comment (rawadabu) [inline projboard/tests/test_client.py]: Please consider testing the template of the page itself, and `my-articles` contains the correct number of articles

**Review by rawadabu — CHANGES_REQUESTED** [review]

File: projboard/tests/test_client.py (hunk 1 above)
Comment (finziamit) [reply projboard/tests/test_client.py]: Added template testing at [0ad48e0](https://github.com/redhat-beyond/Smarticle/pull/87/commits/0ad48e0e5f0009f123062371689e7ff61188d84a)
I don't think `my-articles` need additional tests since the articles are already tested in `test_article.py`
WDYT?

Comment (rawadabu) [reply projboard/tests/test_client.py]: Sounds great then, I think its useless to test twice

File: projboard/templates/myArticles/my_articles.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+<head>
+  <!-- Required meta tags -->
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+  <!-- Bootstrap CSS -->
+  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0/dist/css/bootstrap.min.css" rel="stylesheet"
+    integrity="sha384-gH2yIJqKdNHPEq0n4Mqa/HGKIhSkIHeL5AyhkYV8i59U5AR6csBvApHHNl/vI1Bx" crossorigin="anonymous">
```
Comment (Yarboa) [inline projboard/templates/myArticles/my_articles.html]: Please refer this one
I think it is not needed here
https://github.com/redhat-beyond/Smarticle/pull/87/files#diff-dd3de13a897278d74a566a641962a9569819a02feebf231aadca71d2dcaa1e86R12

File: projboard/templates/myArticles/my_articles.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  <!-- Smarticle CSS -->
+  <link rel="stylesheet" type="text/css" href="{% static 'css/base.css' %}">
+  <link rel="stylesheet" type="text/css" href="{% static 'css/my_articles.css' %}">
```
Comment (Yarboa) [inline projboard/templates/myArticles/my_articles.html:9]: Please refer this 
https://github.com/redhat-beyond/Smarticle/pull/87/files#diff-dd3de13a897278d74a566a641962a9569819a02feebf231aadca71d2dcaa1e86R16

Why there is a need to copy again?

File: projboard/views.py [hunk 4]
```diff
@@ -51,6 +51,24 @@ def create_article(request):
...
+    num_articles = len(my_articles)
+
+    if request.method == "GET":
+        return render(request, 'myArticles/my_articles.html', {
+            'my_articles': my_articles,
+            'num_articles': num_articles})
+    else:
+        raise Http404()
+
+    return render(request, 'myArticles/my_articles.html', {
```
Comment (Anas-mhameed) [inline projboard/views.py]: Do we need this line ??

Comment (Yarboa) [inline projboard/views.py]: Is this code needed? 
you have else with 404,
and then you return the articles. can you explain?

Comment (finziamit) [reply projboard/views.py]: Definitely not. Fixed

File: projboard/templates/myArticles/my_articles.html (hunk 3 above)
Comment (finziamit) [reply projboard/templates/myArticles/my_articles.html:9]: Done

**Review by Shalevro2 — APPROVED** [review]
Outstanding! LGTM

**Review by rawadabu — APPROVED** [review]
LGTM

File: projboard/templates/myArticles/my_articles.html [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+{% block content %}
+<!doctype html>
+<html lang="en">
+
+<head>
+  <!-- Required meta tags -->
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
```
Comment (yaelnoyman1) [inline projboard/templates/myArticles/my_articles.html]: What is found from line 9 to 11, is found in base.html and in my opinion it is not necessary here either

**Review by yaelnoyman1 — CHANGES_REQUESTED** [review]

File: projboard/templates/myArticles/my_articles.html (hunk 5 above)
Comment (finziamit) [reply projboard/templates/myArticles/my_articles.html]: done at [3e5e181](https://github.com/redhat-beyond/Smarticle/pull/87/commits/3e5e181b5adfe2ea1d8c147e8ee20d57c25a89a9)

**Review by yaelnoyman1 — APPROVED** [review]
LGTM

File: projboard/tests/test_client.py (hunk 1 above)
Comment (Anas-mhameed) [reply projboard/tests/test_client.py]: but we should consider checking the return result once we have authentication.

**Review by Anas-mhameed — APPROVED** [review]
LGTM

File: projboard/tests/test_client.py (hunk 1 above)
Comment (Yarboa) [reply projboard/tests/test_client.py]: Please check those returned Values

Comment (finziamit) [reply projboard/tests/test_client.py]: @Yarboa 

I can't do it without the authentication. Opened new issue #95 for it when the authentication will be ready

Comment (Yarboa) [reply projboard/tests/test_client.py]: Sorry, currently there is not authentication
Why cant you receive returned errors?

Comment (finziamit) [reply projboard/tests/test_client.py]: Do you mean to test that the length of my_articles equals num_articles?
Like to add: `assert len(set(response.context["my_articles"])) == response.context[COUNT]`?

Comment (Yarboa) [reply projboard/tests/test_client.py]: Yes, you return something from the view, isnt it? 
And still you can work with users without authentication
Why is it so important?

Comment (finziamit) [reply projboard/tests/test_client.py]: Thought you meant something else.
This can be (and now has been) done without authentication.

Comment (Yarboa) [reply projboard/tests/test_client.py]: You can use this fixture
https://github.com/redhat-beyond/Smarticle/blob/main/projboard/tests/conftest.py#L81

And compare it with the count returned also

Comment (finziamit) [reply projboard/tests/test_client.py]: This fixture counts all the articles in the DB. I test only the articles written by a specific user

Comment (Yarboa) [reply projboard/tests/test_client.py]: OK, so why cant you add a user name that you can grab through the get response.

Once you send user name, like 
client.get("/my_articles/test_user")

This case could be tested, but for the search UI,
The page will not be operatable, unless you add the user name

This way, you could have fixture per user you parse as parameter
Count the article and search through UI

@finziamit ^^

Comment (finziamit) [reply projboard/tests/test_client.py]: done

**Comment by Shalevro2** [comment]
> Resolve #77 and #95

I think it will close just issue #77.
try:
Resolve #77 
Resolve  #95

File: projboard/tests/test_client.py (hunk 1 above)
Comment (Yarboa) [reply projboard/tests/test_client.py]: Please see next comment
Two different tests are better then 2 in one, it is confusing

File: projboard/tests/test_client.py [hunk 6]
```diff
@@ -97,14 +98,26 @@ def test_fill_article_delete(client):
 
 
 @pytest.mark.django_db
-def test_get_my_articles(client):
+def test_get_my_articles(client, user, user_articles):
+    # test edge case of sending user nickname = ''
     response = client.get("/my_articles/")
+    assert response.status_code == 404
```
Comment (Yarboa) [inline projboard/tests/test_client.py]: This should move to different test
But it seems you really understand the point right now
Thanks

Please change this

Comment (finziamit) [reply projboard/tests/test_client.py]: done
